programs/client/Client.cpp

```diff
@@ -1188,6 +1188,9 @@ void Client::processConfig()
     /// - --queries-file command line option is present.
     ///   The value of the option is used as file with query (or of multiple queries) to execute.
 
+    if (config().has("query") && config().has("queries-file"))
+        throw Exception(ErrorCodes::BAD_ARGUMENTS, "Specify either `query` or `queries-file` option");
```

Comment:
`Options '--query' and '--queries-file' cannot be specified at the same time`.

(same in LocalServer.cpp)

Reply:
(I first thought that we should perhaps let Boost program_options do the check (similar to [this](https://stackoverflow.com/a/21914921)) but the disadvantage of that would be that we need to check against the short form `-q` as well. That isn't the case here as the options are translated to configuration already. But I would appreciate if you could add tests for the short form.)

Reply:
No need to check against short form of option in boost variables map (boost::program_options::variables_map). Arguments are already translated to long form after parsing.

Short arguments require check between parsing and storing into variables map, or even before boost options parsing.

Part of code from BaseClient.cpp
```cpp
    auto parser = po::command_line_parser(arguments)
                      .options(options_description.main_description.value())
                      .extra_parser(OptionsAliasParser(options_description.main_description.value()))
                      .allow_unregistered();
    po::parsed_options parsed = parser.run();
 
   // ...
   // po::parsed_options parsed can include both long and short forms of option
   // ...
   
    po::store(parsed, options);
   
   // after po::store options variables map will contain long form arguments
```

I will make test work both with short and long --query option variants

Reply:
Done.